packages/ag-charts-community/src/dom/domManager.ts

```diff
@@ -240,14 +254,15 @@ export class DOMManager extends BaseManager<Events['type'], Events> {
         //
         // Note we do this before relocating the new container to avoid temporarily adding new styles to the page,
         // which may cause a style recalculation
-        if (!isShadowDom) {
+        if (this.documentRoot != null) {
             for (const id of this.rootElements['styles'].children.keys()) {
                 this.removeChild('styles', id);
             }
         }
 
         this.container = pendingContainer;
         this.pendingContainer = undefined;
+        this.documentRoot = this.getShadowDocumentRoot(pendingContainer);
```

Comment:
We did previously support a pathological case where you weren't in a shadow DOM, then you moved into a shadow DOM (and vice-versa). Is this something that still works, or do we not care?

Reply:
I'll verify, if it's an issue we can probably use the `Node.isConnected()` check to know when to be more pessimistic.